src/ds/cli_helper.h

```diff
@@ -19,11 +20,50 @@ namespace cli
     std::string port = {};
   };
 
+  bool parse_address(
+    const std::string_view& addr,
+    ParsedAddress& parsed,
+    const std::string& option_name,
+    const std::string& default_port = "0")
+  {
+    auto found = addr.find_last_of(":");
+    auto hostname = addr.substr(0, found);
+
+    const auto port =
+      found == std::string::npos ? default_port : addr.substr(found + 1);
+
+    // Check if port is in valid range
```

Comment:
I don't think we should effectively forbid the use of service names, they're still very widely supported and sometimes used. If we want to have an integer type internally, we can resolve early for all cases except 0 (https://linux.die.net/man/3/getservbyname).

Reply:
This is just a move/rewrite of the existing logic, that was previously lines 43-52. We can add service name support if we want, but we didn't have it previously so I think we can make that a separate PR.